RadixWallet/Features/AssetTransferFeature/Components/TransferAccountList/TransferAccountList+Reducer.swift

```diff
@@ -290,28 +286,43 @@ extension TransferAccountList {
 		return .none
 	}
 
-	private func determineAdditionalRequiredSignatures(
-		_ receivingAccount: ReceivingAccount.State,
-		forAssets assets: IdentifiedArrayOf<ResourceAsset.State>
-	) -> Effect<Action> {
-		if case let .profileAccount(value: userOwnedAccount) = receivingAccount.recipient {
+	private func signaturesStatusEffect(_ state: State, receivingAccountId: ReceivingAccount.State.ID) -> Effect<Action> {
+		guard
+			let receivingAccount = state.receivingAccounts[id: receivingAccountId],
+			let recipient = receivingAccount.recipient
+		else {
+			return .none
+		}
+
+		typealias DepositStatus = ResourceAsset.State.DepositStatus
+		switch recipient {
+		case let .profileAccount(account):
+			return .run { send in
+				await send(.internal(.setAllDepositStatus(accountId: receivingAccountId, status: .loading)))
```

Comment:
Shouldn't you also check if the `!resourceAddresses.isEmpty` before calling this? the same way you did for externalAccount.